openmetadata-ui/src/main/resources/ui/src/components/GlobalSetting/GlobalSettingLeftPanel.tsx

```diff
@@ -13,61 +13,101 @@
+import {
+  getGlobalSettingMenuItem,
+  getGlobalSettingsMenuWithPermission,
+  MenuList,
+} from '../../utils/GlobalSettingsUtils';
 import { getSettingPath } from '../../utils/RouterUtils';
+import { showErrorToast } from '../../utils/ToastUtils';
+import Loader from '../Loader/Loader';
 import { usePermissionProvider } from '../PermissionProvider/PermissionProvider';
-import { ResourceEntity } from '../PermissionProvider/PermissionProvider.interface';
+import {
+  ResourceEntity,
+  UIPermission,
+} from '../PermissionProvider/PermissionProvider.interface';
 
 const GlobalSettingLeftPanel = () => {
+  const history = useHistory();
   const { tab, settingCategory } = useParams<{ [key: string]: string }>();
+  const [settingResourcePermission, setSettingResourcePermission] =
+    useState<UIPermission>({} as UIPermission);
 
-  const { permissions } = usePermissionProvider();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
 
-  const viewAllPermission = checkPermission(
-    Operation.ViewAll,
-    ResourceEntity.ALL,
-    permissions
-  );
+  const { getResourcePermission } = usePermissionProvider();
 
-  const history = useHistory();
-  const items: ItemType[] = GLOBAL_SETTINGS_MENU.filter(({ isProtected }) => {
-    if (viewAllPermission) {
-      return viewAllPermission;
+  const fetchResourcesPermission = async (resource: ResourceEntity) => {
+    setIsLoading(true);
+    try {
+      const response = await getResourcePermission(resource);
+      setSettingResourcePermission((prev) => ({
+        ...prev,
+        [resource]: response,
+      }));
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    } finally {
+      setIsLoading(false);
     }
+  };
 
-    return !isProtected;
-  }).map(({ category, items }) => {
-    return getGlobalSettingMenus(
-      category,
-      camelCase(category),
-      '',
-      '',
-      items,
-      'group',
-      viewAllPermission
-    );
-  });
+  const menuItems: ItemType[] = useMemo(
+    () =>
+      getGlobalSettingsMenuWithPermission(settingResourcePermission).reduce(
+        (acc: ItemType[], curr: MenuList) => {
+          const menuItem = getGlobalSettingMenuItem(
+            curr.category,
+            camelCase(curr.category),
+            '',
+            '',
+            curr.items,
+            'group'
+          );
+          if (menuItem.children?.length) {
+            return [...acc, menuItem];
+          } else {
+            return acc;
+          }
+        },
+        [] as ItemType[]
+      ),
+    [setSettingResourcePermission]
+  );
 
   const onClick: MenuProps['onClick'] = (e) => {
     // As we are setting key as "category.option" and extracting here category and option
     const [category, option] = e.key.split('.');
     history.push(getSettingPath(category, option));
   };
 
+  useEffect(() => {
+    GLOBAL_SETTING_PERMISSION_RESOURCES.forEach((resource) => {
```

Comment:
This will make number of API calls. Need to check with @harshach if we can reduce it somehow